### ISSUE TITLE
bug: undefined symbol bshin

### ISSUE THREAD

**Issue by OmeletWithoutEgg** [z-shell/zi#124]
### Environment

OSTYPE=linux-gnu CPUTYPE=x86_64 / MACHINE_TYPE=x86_64 ZSH_VERSION=5.8.1

### Expected behavior

Loads the zi/zpmod module successfully.

### Current behavior

After `zmodload zi/zpmod`, it only gives the below error message.
```
zsh: failed to load module `zi/zpmod': /home/user/.zi/zmodules/zpmod/Src/zi/zpmod.so: undefined symbol: bshin
```

### Steps to reproduce

There is a command `zi module build`.
After running it it hints that I can use
```
module_path+=( $HOME/.zi/zmodules/zpmod/Src )
zmodload zi/zpmod
```
to load this module.

### Code snippet

```zsh
### ~/.zshrc

if [[ ! -f $HOME/.zi/bin/zi.zsh ]]; then
  print -P "%F{33}▓▒░ %F{160}Installing (%F{33}z-shell/zi%F{160})…%f"
  command mkdir -p "$HOME/.zi" && command chmod g-rwX "$HOME/.zi"
  command git clone -q --depth=1 --branch "main" https://github.com/z-shell/zi "$HOME/.zi/bin" && \
    print -P "%F{33}▓▒░ %F{34}Installation successful.%f%b" || \
    print -P "%F{160}▓▒░ The clone has failed.%f%b"
fi
source "$HOME/.zi/bin/zi.zsh"
autoload -Uz _zi
(( ${+_comps} )) && _comps[zi]=_zi

module_path+=( $HOME/.zi/zmodules/zpmod/Src )
zmodload zi/zpmod
```

### Aditional information

The result of `cd ~/.zi/zmodules/zpmod; make clean; make`

```sh
cd Src && make prefix='/usr/local' exec_prefix='/usr/local' bindir='/usr/local/bin' libdir='/usr/local/lib' MODDIR='/usr/local/lib/zsh/4.3.17' infodir='/usr/local/share/info' mandir='/usr/local/share/man' datadir='/usr/local/share' fndir='/usr/local/share/zsh/4.3.17/functions' htmldir='/usr/local/share/zsh/htmldoc' CC='gcc' CPPFLAGS='-I/usr/local/include' DEFS='-DHAVE_CONFIG_H' CFLAGS='-g -Wall -O3' LDFLAGS='-L/usr/local/lib' EXTRA_LDFLAGS='-rdynamic' DLCFLAGS='-fPIC' DLLDFLAGS='-shared' LIBLDFLAGS='' EXELDFLAGS='' LIBS='-ldl -lncursesw -lrt -lm  -lc' DL_EXT='so' DLLD='gcc' AWK='gawk' ANSI2KNR=': ansi2knr' YODL=': yodl ' YODL2TXT=': yodl2txt' YODL2HTML=': yodl2html' FUNCTIONS_INSTALL='' tzsh='zsh' all
make[1]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
make[2]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
gawk -f ./signames1.awk  /usr/include/bits/signum-arch.h /usr/include/bits/signum-generic.h >sigtmp.c
case "`gcc -E --version </dev/null 2>&1`" in \
*"Free Software Foundation"*) \
gcc -E -P sigtmp.c >sigtmp.out;; \
*) \
gcc -E sigtmp.c >sigtmp.out;; \
esac
gawk -f ./signames2.awk sigtmp.out > signames.c
rm -f sigtmp.c sigtmp.out
gawk -f ../Src/makepro.awk signames.c Src > signames.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk builtin.c Src > builtin.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk module.c Src > module.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk lex.c Src > lex.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk exec.c Src > exec.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk mem.c Src > mem.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk string.c Src > string.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk parse.c Src > parse.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk hashtable.c Src > hashtable.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk init.c Src > init.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk input.c Src > input.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk loop.c Src > loop.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk utils.c Src > utils.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk params.c Src > params.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk options.c Src > options.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk signals.c Src > signals.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk pattern.c Src > pattern.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk prompt.c Src > prompt.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk compat.c Src > compat.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk jobs.c Src > jobs.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
gawk -f ../Src/makepro.awk glob.c Src > glob.syms
gawk: ../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
make[2]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
make[2]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
grep 'define.*SIGCOUNT' signames.c > sigcount.h
make[3]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
make[3]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
Updated `zsh.mdh'.
echo 'timestamp for zsh.mdh against zsh.mdd' > zsh.mdhs
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < signames.syms) \
	> signames.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < signames.syms) \
	> `echo signames.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < builtin.syms) \
	> builtin.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < builtin.syms) \
	> `echo builtin.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < module.syms) \
	> module.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < module.syms) \
	> `echo module.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < lex.syms) \
	> lex.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < lex.syms) \
	> `echo lex.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < exec.syms) \
	> exec.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < exec.syms) \
	> `echo exec.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < mem.syms) \
	> mem.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < mem.syms) \
	> `echo mem.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < string.syms) \
	> string.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < string.syms) \
	> `echo string.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < parse.syms) \
	> parse.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < parse.syms) \
	> `echo parse.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < hashtable.syms) \
	> hashtable.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < hashtable.syms) \
	> `echo hashtable.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < init.syms) \
	> init.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < init.syms) \
	> `echo init.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < input.syms) \
	> input.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < input.syms) \
	> `echo input.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < loop.syms) \
	> loop.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < loop.syms) \
	> `echo loop.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < utils.syms) \
	> utils.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < utils.syms) \
	> `echo utils.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < params.syms) \
	> params.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < params.syms) \
	> `echo params.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < options.syms) \
	> options.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < options.syms) \
	> `echo options.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < signals.syms) \
	> signals.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < signals.syms) \
	> `echo signals.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < pattern.syms) \
	> pattern.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < pattern.syms) \
	> `echo pattern.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < prompt.syms) \
	> prompt.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < prompt.syms) \
	> `echo prompt.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < compat.syms) \
	> compat.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < compat.syms) \
	> `echo compat.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < jobs.syms) \
	> jobs.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < jobs.syms) \
	> `echo jobs.epro | sed 's/\.epro$/.pro/'`
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < glob.syms) \
	> glob.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < glob.syms) \
	> `echo glob.epro | sed 's/\.epro$/.pro/'`
make[3]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
make[3]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
Updated `zsh.mdh'.
make -f Makemod prefix='/usr/local' exec_prefix='/usr/local' bindir='/usr/local/bin' libdir='/usr/local/lib' MODDIR='/usr/local/lib/zsh/4.3.17' infodir='/usr/local/share/info' mandir='/usr/local/share/man' datadir='/usr/local/share' fndir='/usr/local/share/zsh/4.3.17/functions' htmldir='/usr/local/share/zsh/htmldoc' CC='gcc' CPPFLAGS='-I/usr/local/include' DEFS='-DHAVE_CONFIG_H' CFLAGS='-g -Wall -O3' LDFLAGS='-L/usr/local/lib' EXTRA_LDFLAGS='-rdynamic' DLCFLAGS='-fPIC' DLLDFLAGS='-shared' LIBLDFLAGS='' EXELDFLAGS='' LIBS='-ldl -lncursesw -lrt -lm  -lc' DL_EXT='so' DLLD='gcc' AWK='gawk' ANSI2KNR=': ansi2knr' YODL=': yodl ' YODL2TXT=': yodl2txt' YODL2HTML=': yodl2html' FUNCTIONS_INSTALL='' tzsh='zsh' prep
make[3]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
creating Src/zi/Makefile.in
config.status: creating Src/zi/Makefile
make[4]: Entering directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[4]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[3]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
echo 'timestamp for *.mdd files' > ../Src/modules.stamp
make[3]: Entering directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[4]: Entering directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[4]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src/zi'
Updated `zpmod.mdh'.
echo 'timestamp for zpmod.mdh against zpmod.mdd' > zpmod.mdhs
gawk -f ../../Src/makepro.awk zpmod.c Src/zi > zpmod.syms
gawk: ../../Src/makepro.awk:124: warning: regexp escape sequence `\=' is not a known regexp operator
(echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < zpmod.syms) \
	> zpmod.epro
(echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < zpmod.syms) \
	> `echo zpmod.epro | sed 's/\.epro$/.pro/'`
make[4]: Entering directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[4]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src/zi'
Updated `zpmod.mdh'.
make[3]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[2]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
make[2]: Entering directory '/home/user/.zi/zmodules/zpmod/Src'
make[3]: Entering directory '/home/user/.zi/zmodules/zpmod/Src/zi'
gcc -c -I. -I../../Src -I../../Src -I../../Src/Zle -I. -I/usr/local/include -DHAVE_CONFIG_H -DMODULE -g -Wall -O3 -fPIC -o zpmod..o zpmod.c
zpmod.c: In function ‘bin_readarray’:
zpmod.c:1301:33: warning: variable ‘oarr_name’ set but not used [-Wunused-but-set-variable]
 1301 |         char *callback = NULL, *oarr_name = NULL; // unused: **oarr = NULL;
      |                                 ^~~~~~~~~
zpmod.c:1301:15: warning: variable ‘callback’ set but not used [-Wunused-but-set-variable]
 1301 |         char *callback = NULL, *oarr_name = NULL; // unused: **oarr = NULL;
      |               ^~~~~~~~
zpmod.c:1300:93: warning: variable ‘quantum’ set but not used [-Wunused-but-set-variable]
 1300 |         int delim = '\n', to_copy = 0, start_at = 1, skip_first = 0, remdel = 0, srcfd = 0, quantum = 5000;
      |                                                                                             ^~~~~~~
zpmod.c:1300:70: warning: variable ‘remdel’ set but not used [-Wunused-but-set-variable]
 1300 |         int delim = '\n', to_copy = 0, start_at = 1, skip_first = 0, remdel = 0, srcfd = 0, quantum = 5000;
      |                                                                      ^~~~~~
zpmod.c:1300:54: warning: variable ‘skip_first’ set but not used [-Wunused-but-set-variable]
 1300 |         int delim = '\n', to_copy = 0, start_at = 1, skip_first = 0, remdel = 0, srcfd = 0, quantum = 5000;
      |                                                      ^~~~~~~~~~
zpmod.c:1300:40: warning: variable ‘start_at’ set but not used [-Wunused-but-set-variable]
 1300 |         int delim = '\n', to_copy = 0, start_at = 1, skip_first = 0, remdel = 0, srcfd = 0, quantum = 5000;
      |                                        ^~~~~~~~
zpmod.c:1300:27: warning: variable ‘to_copy’ set but not used [-Wunused-but-set-variable]
 1300 |         int delim = '\n', to_copy = 0, start_at = 1, skip_first = 0, remdel = 0, srcfd = 0, quantum = 5000;
      |                           ^~~~~~~
zpmod.c:1300:13: warning: variable ‘delim’ set but not used [-Wunused-but-set-variable]
 1300 |         int delim = '\n', to_copy = 0, start_at = 1, skip_first = 0, remdel = 0, srcfd = 0, quantum = 5000;
      |             ^~~~~
At top level:
zpmod.c:1670:1: warning: ‘zp_createhashparam’ defined but not used [-Wunused-function]
 1670 | zp_createhashparam(char *name, int flags)
      | ^~~~~~~~~~~~~~~~~~
rm -f zpmod.so
gcc -L/usr/local/lib  -shared -o zpmod.so   zpmod..o    -ldl -lncursesw -lrt -lm  -lc 
make[3]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src/zi'
make[2]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
make[1]: Leaving directory '/home/user/.zi/zmodules/zpmod/Src'
```

**Comment by ss-o** [comment]
Hey @OmeletWithoutEgg, 

I thought it will be an easy fix, however, after fixing it multiple times it still runs into an error. This module is considered deprecated and planned to be removed. :recycle: 

Source moved here: https://github.com/z-shell/zpmod :ballot_box_with_check: 
Additionally issues: https://github.com/z-shell/zpmod/issues/10#issue-1182615162 :space_invader: 

Taking all into consideration it seems there is no reason nor worth spending additional time on this module. I am pretty sure that same could be achieved with ZI, if not I have a whole bunch of new code which can be prioritized. :octocat: 

If I am missing out and you disagree or would like to discuss how we could make a progress on your case I can move this issue to discussions or you can simply drop a message at: 

https://matrix.to/#/#zi:matrix.org :speak_no_evil:

**Comment by ss-o** [comment]
Hey @OmeletWithoutEgg,

The zpmod module extends ZI internally (and could be useful).

My mistake, this issue will be fixed with #137. :octocat:

**Comment by ss-o** [comment]
- https://z.digitalclouds.dev/zh-Hans :cn: 
Support your language: https://digitalclouds.crowdin.com/z-shell :globe_with_meridians:
